packages/common/src/project/readers/local-reader.ts

```diff
@@ -9,14 +10,19 @@ import {IPackageJson} from 'package-json-type';
 import {extensionIsYamlOrJSON} from '../../project';
 
 export class LocalReader implements Reader {
-  constructor(private readonly projectPath: string, private readonly manifestPath: string) {}
+  constructor(
+    private readonly projectPath: string,
+    private readonly manifestPath: string
+  ) {}
 
   get root(): string {
     return path.resolve(this.projectPath);
   }
 
   async getPkg(): Promise<IPackageJson | undefined> {
-    return yaml.load(await this.getFile('package.json')) as IPackageJson;
+    const pkg = (await this.getFile('package.json')) || '';
+    assert(pkg, 'package.json not found');
```

Comment:
I think rather than assert here, we could update `getFile` to no longer return undefined and instead throw an error

Reply:
```
  getProjectSchema(): Promise<unknown | undefined>;
  getPkg(): Promise<IPackageJson | undefined>;
  getFile(file: string): Promise<string | undefined>;
```
Should I modify Reader? Not returning undefined.

Reply:
Yes i think so

Reply:
fix